File: CivitasJava/src/civitas/SorpresaPorJugador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;
/**
 *
 * @author luis
 */

public class SorpresaPorJugador extends Sorpresa{
    private int valor;
    
    SorpresaPorJugador(int valor){
        super("SorpresaPorJugador");
        this.valor = valor;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            SorpresaPagarCobrar pagar = new SorpresaPagarCobrar(this.valor * -1);
            for(int i = 0; i < todos.size(); i++){
                if(i != actual){
                    pagar.aplicarAJugador(i, todos);
                }
            }
            SorpresaPagarCobrar cobrar = new SorpresaPagarCobrar(this.valor * (todos.size() - 1));
            cobrar.aplicarAJugador(actual, todos);
        } 
    }
    
    @Override
    public String toString(){
        String e = "\nNombre: SorpresaPorJugador";
        return e;
    }
}

File: CivitasRuby/lib/sorpresa_por_jugador.rb
# encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaPorJugador < Sorpresa
    
    def initialize(valor)
      super("SorpresaPorJugador")
      @valor = valor
    end
    
       
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        pagar = Sorpresa.new_SORPRESA(Civitas::Tipo_sorpresa::PAGARCOBRAR, @valor * -1, "Pagan todos los jugadores")
        for i in todos
          if i.nombre != todos[actual].nombre
            pagar.aplicar_a_jugador(i,todos)
          end
        end
        cobrar = Sorpresa.new_SORPRESA(Civitas::Tipo_sorpresa::PAGARCOBRAR,@valor * ( todos.lenght - 1), "Cobra de todos los jugadores")
        cobrar.aplicar_a_jugador(actual, todos)
      end
    end
    
         
    def to_s
      "\nNombre: #{@texto}"
    end
    
       
    public_class_method  :new
    
  end
end
